src/main/java/com/sap/sgs/phosphor/fosstars/data/github/GitHubDataFetcher.java

```diff
@@ -405,27 +384,28 @@ protected Date expiration() {
    *
    * @throws IOException If something went wrong.
    */
-  protected void storeLocalRepositoriesInfo() throws IOException {
-    synchronized (LOCAL_REPOSITORIES_INFO_LOCK) {
+  protected static void storeLocalRepositoriesInfo() throws IOException {
+    synchronized (LOCAL_REPOSITORIES_INFO) {
       Files.write(
-          base.resolve(DEFAULT_LOCAL_REPOSITORIES_INFO_FILE),
-          MAPPER.writeValueAsBytes(localRepositoriesInfo));
+          DEFAULT_DIRECTORY.resolve(DEFAULT_LOCAL_REPOSITORIES_INFO_FILE),
+          MAPPER.writeValueAsBytes(LOCAL_REPOSITORIES_INFO));
     }
   }
 
   /**
    * Cleans up local repositories.
    *
    * @param strategy Decides which repositories need to cleaned up.
+   * @throws IOException If something goes wrong.
    */
-  public void cleanup(CleanupStrategy strategy) {
+  public void cleanup(CleanupStrategy strategy) throws IOException {
```

Comment:
I think this method should also clean `LOCAL_REPOSITORIES` up if a local repository is removed. Could you please address this or create an issue for that?

Reply:
The purpose of `cleanup` is to remove the files right? Since `LOCAL_REPOSITORIES` is used as an overall cache. I am not sure if the same purpose applies here. Maybe we should rename this cache?

Reply:
> The purpose of `cleanup` is to remove the files right?

Right. If a local repository is removed, then the elements in `LOCAL_REPOSITORIES`, which point to the removed repository, should be remove as well. It's not about renaming. `LOCAL_REPOSITORIES` may contain invalid items if the files have been removed.

Reply:
I can open an issue for that. Let me know.

Reply:
I will try to address it. If not will create an issue.

Reply:
https://github.com/SAP/fosstars-rating-core/issues/251